### PR TITLE
continue e2e fixing for bamboo

### PR DESCRIPTION
**Pull request by actionless**
still some testcases are executed non-sequentially and sometimes that became noticeable when io is slow

TODO: use request-promise and make helpers with callbacks to return promises

### CODE REVIEW

**Comment by amagdas** [comment]
:+1:

File: client/spec/fetch_spec.js [hunk 1]
```diff
@@ -77,19 +81,26 @@ describe('Fetch', function() {
...
         expect(element.all(by.repeater('items._items')).count()).toBe(2);
 
-        browser.get('/#/workspace/ingest');
-        workspace.switchToDesk('SPORTS DESK');
-        content.setListView();
-        content.actionOnItem('Fetch', 0);
-        browser.get('/#/workspace/content');
-        workspace.switchToDesk('SPORTS DESK');
-        content.setListView();
+        browser.get('/#/workspace/ingest').then(function() {
```
Comment (ioanpocol) [inline client/spec/fetch_spec.js]: Based on the documentation from here: 
http://angular.github.io/protractor/#/control-flow
it is not needed to add "then" after every statement

Comment (actionless) [inline client/spec/fetch_spec.js]: between `expect` command all promises can be resolved unordered

Comment (ioanpocol) [inline client/spec/fetch_spec.js]: I think no. Based on documentation all statements are ordered in control flow queue, not only expect statements.

Comment (actionless) [inline client/spec/fetch_spec.js]: but from screenshots and logs it seems like that happens somehow

Comment (ioanpocol) [inline client/spec/fetch_spec.js]: We have to find the reason. I think that the code became too ugly by adding all those "then" statements

Comment (actionless) [inline client/spec/fetch_spec.js]: it was but the thing what no one of developers which were trying this issue before managed to find the reason of that "desynchronization", so if i won't manage to find the core problem of that than maybe we can consider moving to angular-independed e2e framework, like http://nightwatchjs.org/ or http://dalekjs.com/pages/getStarted.html

**Comment by actionless** [comment]
can't reproduce the failure after that kernel tuning, let's see how it will be over the time

**Comment by ioanpocol** [comment]
If after a few days it is working we should rolback "then" changes and check again for few days if it is working

**Comment by actionless** [comment]
@ioanpocol master branch was still failing after write buffer was reduced, so it seems like they those fixes are work together

**Pull request merged**